packages/harmony/src/components/NavItem/NavItem.tsx

```diff
@@ -0,0 +1,90 @@
+import { useTheme } from '@emotion/react'
+
+import { Flex } from '../layout/Flex'
+import { Text } from '../text/Text'
+
+import type { NavItemProps } from './types'
+
+/**
+ * A navigation item component, typically used in sidebars or navigation menus.
+ * Supports default, hover, and selected states, as well as optional left and right icons.
+ */
+export const NavItem = ({
+  children,
+  leftIcon: LeftIcon,
+  rightIcon: RightIcon,
+  isSelected = false,
+  ...props
+}: NavItemProps) => {
+  const { color } = useTheme()
+
+  const hasLeftIcon = !!LeftIcon
+  const hasRightIcon = !!RightIcon
+
+  const backgroundColor = isSelected ? color.secondary.s400 : undefined
+
+  const textColor = isSelected ? 'staticWhite' : 'default'
+
+  const iconColor = isSelected ? color.static.staticWhite : color.neutral.n800
+
+  return (
+    <Flex
+      alignItems='center'
+      gap='s'
+      pl='s'
+      pr='s'
+      css={{
+        width: '240px',
+        cursor: 'pointer',
+        transition: 'background-color 0.18s ease-in-out'
+      }}
+      {...props}
+    >
+      <Flex
+        alignItems='center'
+        flex={1}
+        gap='m'
+        p='s'
+        borderRadius='m'
+        css={{
+          backgroundColor,
+          borderWidth: '1px',
+
+          '&:hover': {
+            backgroundColor: isSelected ? undefined : color.background.surface2,
+            borderColor: color.border.default
+          }
+        }}
+      >
+        <Flex alignItems='center' gap='m' flex={1}>
+          {hasLeftIcon && (
+            <LeftIcon
+              height={24}
```

Comment:
for icons, we have props you can pass it for size, check out the height chart here: https://harmony.audius.co/?path=/docs/icons-utility-icons--documentation, you can also pass "color" prop similar to text, tokens defined here: https://github.com/AudiusProject/audius-protocol/blob/14f63c712781d63bba19c6aca5a61429dc4da517/packages/harmony/src/foundations/color/semanticV2.ts#L10 in youre case seems like "inverse" may be the move?